### DISCUSSION TITLE
Conditional Yup and errors displaying

### DISCUSSION THREAD

**Discussion by rlesniak** [react-hook-form/react-hook-form#3810] [General] [answered]
Strange problem, in Formik it worked perfectly.
Why on checkbox toggle, errors aren't showing? Only when Submit is called, errors object is refreshing

Perfect reproduction with Formik working preview
https://codesandbox.io/s/react-hook-form-validationschema-v6-forked-ou04i?file=/src/index.js

**Comment by bluebill1049** [comment, marked as answer]
because hook form is focused on single input validation to enhance the perf, if you need dependent input to be validated, use `trigger to trigger the other input's validation. hook form doesn't fire entire form validation unless you told so and (render the entire form).

**Reply by rlesniak** [reply]
Got it. Thank you
